### PR TITLE
Adds ctrl+home/end, cmd+home/end key combinations

### PR DESCRIPTION
**Pull request by rutvik110**
This PR adds the following 4 new key combinations to `KeyboardInput` extension, 
`pressCmdHome`, `pressCmdEnd`, `pressCtrlHome`, `pressCtrlEnd`.

### CODE REVIEW

File: lib/src/input_method_engine.dart [hunk 1]
```diff
@@ -192,7 +192,7 @@ class ImeSimulator {
       'method': 'TextInputClient.updateEditingStateWithDeltas',
     });
 
-    await ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+    await _tester.binding.defaultBinaryMessenger.handlePlatformMessage(
```
Comment (matthew-carroll) [inline lib/src/input_method_engine.dart:195]: Why was this changed?

Comment (rutvik110) [reply lib/src/input_method_engine.dart:195]: That method has been deprecated after `v3.9.0-19.0.pre` as mentioned [here](https://api.flutter.dev/flutter/services/BinaryMessenger/handlePlatformMessage.html).

Comment (matthew-carroll) [reply lib/src/input_method_engine.dart:195]: Sounds good. Thanks.
